.github/workflows/get-modules-metadata.yml

```diff
@@ -0,0 +1,40 @@
+name: Get modules metadata
+on:
+  workflow_call:
+    secrets:
+      # Secrets must be passed from the caller workflow explicitly.
+      CLIENT_ID:
+        required: true
+      TENANT_ID:
+        required: true
+      SUBSCRIPTION_ID:
+        required: true
+  workflow_dispatch:
+
+permissions:
+  id-token: write
+  contents: read
```

Comment:
Are the secrets needed? If not, we should remove them.

Reply:
Yes, I'll be sending out a follow up PR to login to azure and write the modules metadata to storage account container. Let me know if you want me to remove it here and include it in other PR.

Reply:
No, in that case we can leave them as is.